packages/mui-codemod/src/v6.0.0/styled/test-cases/BasicStyled.expected.js

```diff
@@ -0,0 +1,104 @@
+const FormHelperTextRoot = styled('p')(({
+  theme
+}) => (({
+  color: (theme.vars || theme).palette.text.secondary,
+  ...theme.typography.caption,
+  textAlign: 'left',
+
+  [`&.${formHelperTextClasses.disabled}`]: {
+    color: (theme.vars || theme).palette.text.disabled,
+  },
+
+  variants: [{
+    props: {
+      size: 'small'
+    },
+
+    style: {
+      marginTop: 4,
+    }
+  }, {
+    props: {
+      variant: 'contained',
+      size: 'small'
+    },
+
+    style: {
+        marginTop: 6,
+      }
+  }, {
+    props: (
+      {
+        disabled,
+        ownerState
+      }
+    ) => ownerState.size === 'small' &&
+      ownerState.variant === 'contained' &&
+      disabled,
+
+    style: {
+        marginTop: 6,
+      }
+  }, {
+    props: (
+      {
+        ownerState
+      }
+    ) => ownerState.size !== 'small',
```

Comment:
I think we decided to now use `ownerState`, as it is already merged on top of the props.

Reply:
The codemod is not responsible for the prop name, it will transform with the prop name it found.

Ex,
```js
styled('div')(({ theme, disabled }) => ({
  …disabled && {
    ...
  }
});
```

will be transformed to
```js
styled('div')(({ theme, disabled }) => ({
  variants: [
    { props: ({ disabled }) => disabled, style: {…} },
  ]
});
```

Also, the reason I don't use object for props is that it's safer to preserve the logic since spreading will apply if the condition is true. Using object could fail in this case:

```js
styled('div')(({ theme, disabled }) => ({
  // Chip
  …onClick && {
    …styles
  }
});
```

Using object is wrong:
```js
styled('div')(({ theme, disabled }) => ({
  variants: [
    { props: { onClick: true }, style: {…} },
  ]
});
```